auth/resource.go

```diff
@@ -2,175 +2,149 @@ package auth
 
 import (
 	"context"
+	"net/http"
+	"net/url"
 
-	"fmt"
-
-	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/auth/authservice"
+	"github.com/fabric8-services/fabric8-wit/goasupport"
 	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/fabric8-services/fabric8-wit/rest"
 	"github.com/goadesign/goa"
-	"github.com/satori/go.uuid"
+	goaclient "github.com/goadesign/goa/client"
+	goauuid "github.com/goadesign/goa/uuid"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
 )
 
-// AuthzResourceManager represents a space resource manager
-type AuthzResourceManager interface {
-	CreateResource(ctx context.Context, request *goa.RequestData, name string, rType string, uri *string, scopes *[]string, userID string) (*Resource, error)
-	DeleteResource(ctx context.Context, request *goa.RequestData, resource Resource) error
-}
-
-// KeycloakResourceManager implements AuthzResourceManager interface
-type KeycloakResourceManager struct {
-	configuration KeycloakConfiguration
+// ResourceManager represents a space resource manager
+type ResourceManager interface {
+	CreateSpace(ctx context.Context, request *goa.RequestData, spaceID string) (*authservice.SpaceResource, error)
+	DeleteSpace(ctx context.Context, request *goa.RequestData, spaceID string) error
 }
 
-// Resource represents a Keycloak resource and associated permission and policy
-type Resource struct {
-	ResourceID   string
-	PolicyID     string
-	PermissionID string
+// AuthzResourceManager implements ResourceManager interface
+type AuthzResourceManager struct {
+	configuration AuthServiceConfiguration
 }
 
-// KeycloakConfiguration represents a keycloak configuration
-type KeycloakConfiguration interface {
-	GetKeycloakEndpointAuthzResourceset(*goa.RequestData) (string, error)
-	GetKeycloakEndpointToken(*goa.RequestData) (string, error)
-	GetKeycloakEndpointClients(*goa.RequestData) (string, error)
-	GetKeycloakEndpointAdmin(*goa.RequestData) (string, error)
-	GetKeycloakEndpointEntitlement(*goa.RequestData) (string, error)
-	GetKeycloakClientID() string
-	GetKeycloakSecret() string
+// AuthServiceConfiguration represents auth service configuration
+type AuthServiceConfiguration interface {
```

Comment:
Could we move this out to somewhere non-resource-ish ? Because `AuthServiceConfiguration` would be used across many services like login,user profile,account linking etc..